src/worker/helpers.go

```diff
@@ -166,46 +166,224 @@ func initOLDir(olPath string, dockerBaseImage string, newBase bool) (err error)
 	return nil
 }
 
-// stop OL if running (or just return if not).
-// main error scenarios:
-// 1. PID exists, but process cannot be killed (worker probably died unexpectedly)
-// 2. The cleanup is taking too long (maybe the timeout is insufficient, or there is a deadlock)
-func stopOL(_ string) error {
-	// locate worker.pid, use it to get worker's PID
+// Define a custom type for the state
+type OlState int
+
+// Define constants for the different states
+const (
+	Uninitialized OlState = iota
+	Running
+	StoppedClean
+	StoppedDirty
+	Unknown = -1
```

Comment:
What should I return in check state if there is an error though? I cannot simply return nil because OlState is a number.
Would you suggest me to simply return -1?

Reply:
Good point!